### DISCUSSION TITLE
How to use vue with rclnodejs create ros2 Web

### DISCUSSION THREAD

**Discussion by naiveZhou** [RobotWebTools/rclnodejs#925] [General]
Hi,i am a AGV developer from China.I have already read the book 'A+Concise+Introduction+to+Robot+Programming+with+ROS2'.Now,i wonder if there is such a way as this link 'https://cxx0822.github.io/2021/10/20/ji-yu-vue.js-de-rosweb-huan-jing-da-jian/'. Ros2 Web can be implemented simply by using vue.js and rclnodejs ?
Looking forward to your reply.

**Comment by minggangw** [comment]
I don't have much experience in setting up a web application using vue.js. As far as I know, Electron should be the right choice for you if you want to leverage rclnodejs, meanwhile, construct the UI by vue.js. The way of programming for it is just like a normal Electron application, importing the nodejs module and transferring the data from rclnodejs to the UI part, e.g., for visualization, hope it helps.
